lib/console/deployments/pr/impl/bitbucket.ex

```diff
@@ -89,11 +125,84 @@ defmodule Console.Deployments.Pr.Impl.BitBucket do
   end
 
   defp get_pull_id(url) do
-    with {:ok, %URI{path: "/" <> path}} <- URI.parse(url),
-         [org, repo, "pull-requests" <> number] <- String.split(path, "/") do
-      {:ok, "#{org}/#{repo}", number}
+    with %URI{path: "/" <> path} <- URI.parse(url),
+         parts <- String.split(path, "/"),
+         [workspace, repo, "pull-requests", pr_id] <- parts do
+      {:ok, workspace, repo, pr_id}
     else
       _ -> {:error, "could not parse bitbucket url"}
     end
   end
+
+  defp extract_api_url(url) do
+    case is_binary(url) and String.trim(url) != "" do
+      true ->
+        url
+      false ->
+        @bitbucket_api_url
+    end
+  end
+
+  defp get_pr_info(conn, workspace, repo, pr_id) do
+    with {:ok, api_url, token} <- url_and_token(conn, @bitbucket_api_url),
+         api_url = extract_api_url(api_url),
+         pr_url = "#{api_url}/repositories/#{workspace}/#{repo}/pullrequests/#{pr_id}",
+         {:ok, pr_response} <- HTTPoison.get(pr_url, [{"Authorization", "Bearer #{token}"}, {"Content-Type", "application/json"}]),
+         {:ok, pr_body} <- Jason.decode(pr_response.body) do
+      {:ok, pr_body}
+    else
+      {:error, %HTTPoison.Error{reason: reason}} ->
+        {:error, "HTTP request failed: #{inspect(reason)}"}
+      {:error, %Jason.DecodeError{}} ->
+        {:error, "Invalid JSON response"}
+      {:error, reason} ->
+        {:error, reason}
+    end
+  end
+
+  defp files_diff_list(diffstat_url) do
+    case HTTPoison.get(diffstat_url) do
```

Comment:
I'm like 99% sure this fetch will need auth, likely use the encapsulated get function for it (true for other fetches too)